common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -103,14 +103,22 @@ public void fetchBlocks(
     try {
       RetryingBlockFetcher.BlockFetchStarter blockFetchStarter =
           (blockIds1, listener1) -> {
-            // Unless this client is closed.
-            if (clientFactory != null) {
-              TransportClient client = clientFactory.createClient(host, port);
-              new OneForOneBlockFetcher(client, appId, execId,
-                blockIds1, listener1, conf, downloadFileManager).start();
-            } else {
-              logger.info("This clientFactory was closed. Skipping further block fetch retries.");
-            }
+           try{
+             // Unless this client is closed.
+             if (clientFactory != null) {
+               TransportClient client = clientFactory.createClient(host, port);
+               new OneForOneBlockFetcher(client, appId, execId,
+                   blockIds1, listener1, conf, downloadFileManager).start();
+             } else {
+               logger.info("This clientFactory was closed. Skipping further block fetch retries.");
+             }
+           } catch (IOException e) {
+             logger.info("The relative remote external shuffle service (host: " + host + "," +
+                 "port: "+ port + "), which maintains the block data can't been connected.");
```

Comment:
so what you do is just printing a log when error happens?

Reply:
> so what you do is just printing a log when error happens?

Sorry, I miss the way to stop retry.

May be we should add some way to check external shuffle service alive?